web/documentserver-example/java/src/main/java/helpers/FileUtility.java

```diff
@@ -19,64 +19,39 @@
 package helpers;
 
 import entities.FileType;
+import format.FormatManager;
+
 import java.net.URL;
-import java.util.Arrays;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 
 public final class FileUtility {
+    private static FormatManager formatManager = new FormatManager();
 
     private FileUtility() { }
 
     // get file type
     public static FileType getFileType(final String fileName) {
         String ext = getFileExtension(fileName).toLowerCase();
 
-        // word type for document extensions
-        if (extsDocument.contains(ext)) {
-            return FileType.Word;
-        }
-
-        // cell type for spreadsheet extensions
-        if (extsSpreadsheet.contains(ext)) {
-            return FileType.Cell;
-        }
-
-        // slide type for presentation extensions
-        if (extsPresentation.contains(ext)) {
-            return FileType.Slide;
+        try {
```

Comment:
```
FileUtility.formatManager.documentExtensions() -> documents() -> all() -> Files.readAllLines();
FileUtility.formatManager.spreadsheetExtensions() -> spreadsheet() -> all() -> Files.readAllLines();
FileUtility.formatManager.presentationExtensions() -> presentation() -> all() -> Files.readAllLines();

```
For one getFileType function call, the file (onlyoffice-docs-formats.json) is read three times.

```
for (Format format : FileUtility.formatManager.all()) {
   if (format.getName().equals(ext)) {
        return format.getType();
   }
}
```